src/Waives.Pipelines/Pipeline.cs

```diff
@@ -276,9 +282,15 @@ void OnDocumentException(Exception exception, Document document)
                 OnPipelineError,
                 _maxConcurrency);
 
-            _docSource.Subscribe(pipelineObserver);
-
-            await taskCompletion.Task;
+            var connection = _docSource.Subscribe(pipelineObserver);
```

Comment:
Flagging for #84. Also `Subscribe` returns an `IDisposable`, so we can use a `using` here instead of `try..finally`.